src/common/services/email/email.service.ts

```diff
@@ -28,7 +28,52 @@ export class EmailService {
     const sendResults = await Promise.all(emailPromises);
     if (isLocalhost) {
       sendResults.forEach((result, index) => {
-        console.log(`📨 Email ${index + 1} sent. Preview URL: ${nodemailer.getTestMessageUrl(result)}`);
+        console.log(
+          `📨 Email ${index + 1} sent. Preview URL: ${nodemailer.getTestMessageUrl(result)}`
+        );
+      });
+    }
+
+    return sendResults;
+  }
+
+  async sendBatchDiscountCodeEmails(
+    emailDataArray: {
+      recipientEmail: string;
+      subject: string;
+      context: {
+        code: string;
+        discountName: string;
+        discountType: "voucher" | "discount";
+        eventName?: string;
+        reusable: boolean;
+        redemptionUrl?: string;
+      };
+    }[],
+    isLocalhost: boolean
+  ) {
+    const emailPromises = emailDataArray.map(
```

Comment:
our EmailService has function `emailService.sendBatchEmails`, it is used already in 2 places, maybe we can use it here as well?